api/src/repositories/summary-repository.ts

```diff
@@ -299,11 +301,11 @@ export class SummaryRepository extends BaseRepository {
 
     const response = await this.connection.query<{ delete_timestamp: string }>(sqlStatement.text, sqlStatement.values);
 
-    if (!response || !response?.rows[0]?.delete_timestamp) {
+    if (!response?.rows[0]?.delete_timestamp) {
       throw new HTTP400('Failed to soft delete survey summary submission');
     }
 
-    return (response && response.rowCount) || null;
+    return response?.rowCount || null;
```

Comment:
super nit: technically we don't need to `?.` the rows or rowCount variable, as they will always be present.

Reply:
(The sonarcloud thing isnt smart enough to know that, it just sees the `||` check and raises the warning that it prefers `?.`)